src/main/kotlin/com/openlattice/data/storage/PostgresDataQueries.kt

```diff
@@ -23,14 +24,20 @@ import java.util.*
 internal class PostgresDataQueries
 
 const val VALUES = "values"
+const val PROPERTIES = "properties"
 
 val dataTableColumnsSql = PostgresDataTables.dataTableColumns.joinToString(",") { it.name }
 
-val valuesColumnsSql = PostgresDataTables.dataTableValueColumns.joinToString(",") {
-    "array_agg(${it.name}) FILTER (where ${it.name} IS NOT NULL) as ${it.name}"
-}
+val detailedValueColumnsSql = dataTableValueColumns.joinToString("||") {
+    "jsonb_agg(json_build_object('value',${it.name},'${ENTITY_SET_ID.name}', entity_set_id, '${ID_VALUE.name}', origin_id)) FILTER (where ${it.name} IS NOT NULL)"
```

Comment:
Seems safer/clearer to use the PostgresColumn names everywhere instead of mixing them with hardcoded values. Also I think this would be a bit easier to read if it were written as 
```
"jsonb_agg(json_build_object(
  'value', ${it.name},
  '${ENTITY_SET_ID.name}', ${ENTITY_SET_ID.name},
  '${ID_VALUE.name}', ${ORIGIN_ID.name}
)) FILTER (
  WHERE ${it.name} IS NOT NULL
)"
```

Reply:
Second